

# Hypothesis Testing Exercise

# **Q1**

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions. Cutlets.csv



*  Null hyposthesis , Ho: μ1 = μ2 (There is no difference in diameters of cutlets between two units).
*   Alternate hypothesis , Ha: μ1 ≠ μ2 (There is significant difference in diameters of cutlets between two units)



In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [ ]:
# Uploading the data
from google.colab import files
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


In [ ]:
# load the dataset
data=pd.read_csv("Cutlets.csv")
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
Unit_A=pd.Series(data.iloc[:,0])
Unit_A

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [ ]:
Unit_B=pd.Series(data.iloc[:,1])
Unit_B

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [ ]:
# use 2-sample 2-tail ttest:
# stats.ttest_ind(array1,array2)
p_value=stats.ttest_ind(Unit_A,Unit_B)
p_value

TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

In [ ]:
# To print the p_value
p_value[1]

0.4722394724599501

Comparing the p_value with level of significance,α = 0.05(at 5%)\
i.e p_value=0.472 > α = 0.05


Accept the Null Hypothesis  
\i.e There is no difference in diameters of cutlets between two units

# **Q2**

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.



*  Anova/F test : which is used to checkif the means of two or more groups are statisticaly different from each other.
*   Null hyposthesis , Ho: μ1 = μ2 = μ3 = μ4(all sample TAT population means are same)\
Alternate hypothesis , Ha : Atleat one the saample TAT population mean is different.



In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# uploading the data
from google.colab import files
uploaded=files.upload()

Saving LabTAT.csv to LabTAT.csv


In [ ]:
# load the dataset
df=pd.read_csv("LabTAT.csv")
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [ ]:
import scipy.stats as stats
p_value=stats.f_oneway(df.iloc[:,0],df.iloc[:,1],df.iloc[:,2],df.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [ ]:
# to print p_value
p_value[1]

2.1156708949992414e-57

Comparing the p_value with level of significance,α = 0.05(at 5%)

i.e p_value=2.1156708949992414e-57 < α = 0.05


Reject the Null Hypothesis

# **Q3**



Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.



*   Chi-Square test :  Chi-Square tests of independence test whether two qualitative variables are independent,i.e whether there exists a relationship between two catagorical variables
*   Null hypothesis: the variables are independent , there is no association or dependency between the gender based buyer rations across regions\
Alternative hypothesis : the variables are dependent , there is significant association or dependency between the gender based buyer rations across regions



In [ ]:
import pandas as pd
from scipy import stats as stats
import numpy as np

In [ ]:
# uploading the data
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


In [ ]:
# load the dataset
df=pd.read_csv("BuyerRatio.csv")
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
df_table=df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
df_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
value=stats.chi2_contingency(df_table)
value

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [ ]:
#degrees of freedom=(no. of rows-1)*(no. of columns-1)
dof=value[2]
dof

3

In [ ]:
Expected_value=value[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [ ]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2/e for o,e in zip(df_table.values,Expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [ ]:
# from chi-square table(α = 0.05 and df=3)
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

test statistic=1.515295 < critical_value=7.8147279\
failed to reject null hypothesis

 Therefore, there is no association or dependency between male-female buyers rations and are similar across regions. Hence, Independent samples

# **Q4**

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.



*   Chi-Square test : Chi-Square tests of independence test whether two qualitative variables are independent,i.e whether there exists a relationship between two catagorical variables
*    Null hypothesis: the variables are independent , customer order forms defective % does not varies by center\
Alternative hypothesis :the variables are dependent , customer order forms defecctive % varies by center



In [ ]:
import pandas as pd
import numpy as np
from scipy import stats as stats
from scipy.stats import chi2_contingency
from scipy.stats import chi2

In [ ]:
from google.colab import files
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm (1).csv


In [ ]:
data=pd.read_csv("Costomer+OrderForm.csv")
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [ ]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
#make a contingency table
observed_fre=np.array([[271,267,269,280],[29,33,31,20]])
observed_fre

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
value=stats.chi2_contingency(observed_fre)
value

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
expected_fre=value[3]
expected_fre

array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]])

In [ ]:
dof=value[2]
dof

3

In [ ]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2/e for o,e in zip(observed_fre
,expected_fre)])
chi_square_statestic=chi_square[0]+chi_square[1]+chi_square[2]+chi_square[3]
chi_square_statestic

3.8589606858203545

In [ ]:
chi_square

array([0.02198142, 0.88169925, 0.295528  , 2.65975202])

In [ ]:
# from chi-square table(α = 0.05 and df=3)
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

test statistic=3.8589606 < critical_value=7.8147279\
failed to reject null hypothesis

customer order forms defective % does not varies by center